openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -652,6 +655,26 @@ def _apply_cdn_to_url(self, url, base_url):
 
         return urlunparse((None, base_url, path, params, query, fragment))
 
+    def get_prefetched_course_mode(self, mode_slug):
+        """
+        Find the specified mode associated with this course.
+
+        Returns:
+            The course mode if available, or None if the mode could not be found or has already expired.
+        """
+        now = datetime.datetime.now(pytz.UTC)
+
+        # In some code paths, the modes have been prefetched. Instead of calling get() directly here, which will
+        # miss the cache, instead just iterate through the list of prefetched modes and find the verified mode.
+        for mode in self.modes.all():
+            if mode.mode_slug == mode_slug:
+                if mode.expiration_datetime is None or now < mode.expiration_datetime:
+                    return mode
+                else:
+                    return None
```

Comment:
Nit (feel free to ignore): Can remove the above 2 lines and the the code would still work the same.